cmd/notifications/main.go

```diff
@@ -63,8 +62,31 @@ func main() {
 
 	connectionsRepository := repositories.NewConnections()
 
+	var vaultCli *vault.Client
+	if cfg.VaultUserPassAuthEnabled {
+		log.Info(ctx, "Vault userpass auth enabled")
+		vaultCli, err = providers.NewVaultClientWithUserPassAuth(ctx, cfg.KeyStore.Address, cfg.VaultUserPassAuthPassword)
+		if err != nil {
+			log.Error(ctx, "cannot init vault client with Kubernetes Auth: ", "err", err)
+			return
+		}
+	} else {
+		log.Info(ctx, "Vault userpass auth not enabled")
+		vaultCli, err = providers.NewVaultClient(cfg.KeyStore.Address, cfg.KeyStore.Token)
+		if err != nil {
+			log.Error(ctx, "cannot init vault client: ", "err", err)
+			return
+		}
+	}
```

Comment:
Same comment as in cmd/issuer_initializer/main.go.   I think we can reuse this code